go-controller/pkg/ovn/obj_retry.go

```diff
@@ -874,36 +1099,48 @@ func (oc *Controller) iterateRetryResources(r *retryObjs, updateAll bool) {
 
 		klog.Infof("%v %s: retry object setup", r.oType, objKey)
 
-		// delete old object if needed
-		if entry.oldObj != nil {
-			klog.Infof("%v retry: removing old object for %s", r.oType, objKey)
-			if !isResourceScheduled(r.oType, entry.oldObj) {
-				klog.V(5).Infof("Retry: %s %s not scheduled", r.oType, objKey)
-				continue
-			}
-			if err := oc.deleteResource(r, entry.oldObj, entry.config); err != nil {
-				klog.Infof("%v retry delete failed for %s, will try again later: %v", r.oType, objKey, err)
+		if resourceNeedsUpdate(r.oType) && entry.oldObj != nil && entry.newObj != nil {
+			klog.Infof("%v retry: updating object %s", r.oType, objKey)
+			if err := oc.updateResource(r, entry.oldObj, entry.newObj); err != nil {
+				klog.Infof("%v retry update failed for %s, will try again later: %v", r.oType, objKey, err)
 				entry.timeStamp = time.Now()
 				continue
 			}
-			// successfully cleaned up old object, remove it from the retry cache
+			// successfully cleaned up new and old object, remove it from the retry cache
+			entry.newObj = nil
 			entry.oldObj = nil
-		}
-
-		// create new object if needed
-		if entry.newObj != nil {
-			klog.Infof("%v retry: creating object for %s", r.oType, objKey)
-			if !isResourceScheduled(r.oType, entry.newObj) {
-				klog.V(5).Infof("Retry: %s %s not scheduled", r.oType, objKey)
-				continue
+		} else {
+			// delete old object if needed
+			if entry.oldObj != nil {
+				klog.Infof("%v retry: removing old object for %s", r.oType, objKey)
+				if !isResourceScheduled(r.oType, entry.oldObj) {
+					klog.V(5).Infof("Retry: %s %s not scheduled", r.oType, objKey)
+					continue
+				}
+				if err := oc.deleteResource(r, entry.oldObj, entry.config); err != nil {
+					klog.Infof("%v retry delete failed for %s, will try again later: %v", r.oType, objKey, err)
+					entry.timeStamp = time.Now()
+					continue
+				}
+				// successfully cleaned up old object, remove it from the retry cache
+				entry.oldObj = nil
 			}
-			if err := oc.addResource(r, entry.newObj, true); err != nil {
-				klog.Infof("%v retry create failed for %s, will try again later: %v", r.oType, objKey, err)
-				entry.timeStamp = time.Now()
-				continue
+
+			// create new object if needed
+			if entry.newObj != nil {
+				klog.Infof("%v retry: creating object for %s", r.oType, objKey)
+				if !isResourceScheduled(r.oType, entry.newObj) {
+					klog.V(5).Infof("Retry: %s %s not scheduled", r.oType, objKey)
+					continue
+				}
+				if err := oc.addResource(r, entry.newObj, true); err != nil {
+					klog.Infof("%v retry create failed for %s, will try again later: %v", r.oType, objKey, err)
+					entry.timeStamp = time.Now()
+					continue
+				}
+				// successfully cleaned up old object, remove it from the retry cache
```

Comment:
Here I didn't read it carefully myself... could you replace old with new in the comment above? thanks :)